### PR TITLE
make duplicate rule return 0 rows and full count

### PR DESCRIPTION
**Pull request by thinhha**
This fixes two bugs with the existing duplicates check rule:

- previously the duplicates rule returned nothing so it did not update the failure count to 0 once the duplicates are removed, with this change it returns 0 as the count
- also it will return the total number of duplicated records not just the distinct count of duplicated records

### CODE REVIEW

File: configs/rules/complex-rules.yml [hunk 1]
```diff
@@ -20,8 +20,13 @@ rules:
       custom_sql_arguments:
         - column_names
       custom_sql_statement: |-
-        select
-          $column_names
-        from data
-        group by $column_names
-        having count(*) > 1
+        select a.*, duplicates.*
```
Comment (andrew-ii) [inline configs/rules/complex-rules.yml:23]: General comment for later consideration (not blocking the overall PR approval) - would be good to review the interface with `custom_sql_statement` queries. Right now it seems to be arbitrary, while I believe it needs to be a set interface to allow managing custom queries under a general framework. Again, not a blocker for this specific change

**Review by andrew-ii — APPROVED** [review]

File: configs/rules/complex-rules.yml (hunk 1 above)
Comment (thinhha) [reply configs/rules/complex-rules.yml:23]: Thanks Andrew. Currently each custom_sql_statement string is substituted into this macro template: https://github.com/GoogleCloudPlatform/cloud-data-quality/blob/main/clouddq/templates/dbt/macros/macros.sql#L39

Apart from the fact that you can only run a single select query, it is pretty much arbitrary at the moment.

Comment (thinhha) [reply configs/rules/complex-rules.yml:23]: I considered updating the macro/rule type itself but it wasn't necessary to get the target outcome for this rule. In theory, once duplicates works properly this rule should generalize to other rule types.

One downside of this approach is that it is not currently possible to retrieve only the failing records. I decided to implement that as part of a separate scope to materialize all failing records not just the complex rules, as that feature will require a few other things to change as well.

Comment (thinhha) [reply configs/rules/complex-rules.yml:23]: any further thoughts let me know otherwise I'll merge and release a patch for the current release.

**Review by AmandeepSinghCS — APPROVED** [review]

**Pull request merged**